db/sequence_id.go

```diff
@@ -179,21 +179,29 @@ func parseClockSequenceID(str string, sequenceHasher *sequenceHasher) (s Sequenc
 			}
 		}
 	} else if len(components) == 2 {
-		// TriggeredBy Clock Hash, and vb.seq sequence
+		// TriggeredBy Clock Hash, and triggerdvb.vb.seq sequence
 		if s.TriggeredByClock, err = sequenceHasher.GetClock(components[0]); err != nil {
 			return SequenceID{}, err
 		}
 		sequenceComponents := strings.Split(components[1], ".")
-		if len(sequenceComponents) != 2 {
+		if len(sequenceComponents) != 3 {
```

Comment:
Can you add a comment why you are getting `components[1]`?  Maybe add a constant like `SEQUENCE_COMPONENT = 1` and then call `components[SEQUENCE_COMPONENT]`

Also, can you add a comment that explains why there should be three sequence components?  Eg, "Must be 3 components since we are in backfill"